### ISSUE TITLE
geopandas.read_file() fails with recursion error

### ISSUE THREAD

**Issue by ccamara** [PyCQA/pylint#2756]
### Steps to reproduce
1. Open Spyder 3.3.3
2. Create a file with the following code:

```python
geopandas.read_file("data/export.geojson")
```
3.  Execute static code analyis.

### Current behavior

I cannot execute Static code analysis on spyder 3.3.3, as it gets stopped and a new black window with "error" title is displayed, but no more information.

If I comment out that line, code analysis work as usual.

### Expected behavior

Code analysis working as usual

### pylint --version output

```
pylint 2.2.2
astroid 2.1.0
Python 3.7.2 (default, Dec 29 2018, 06:19:36) 
[GCC 7.3.0]
```

### Traceback
```python-traceback
RecursionError: maximum recursion depth exceeded

WARNING:spyder.widgets.github.backend:Failed to send bug report on Github. response={'code': 401, 'json': {'message': 'Bad credentials', 'documentation_url': 'https://developer.github.com/v3'}
```

## Versions

* Spyder version: 3.3.3 
* Python version: 3.7.1
* Qt version: 5.6.2
* PyQt5 version: 5.6
* Operating System: Linux 4.15.0-45-generic

### Dependencies

```
pyflakes >=0.6.0  :  2.1.0 (OK)
pycodestyle >=2.3 :  2.5.0 (OK)
pygments >=2.0    :  2.3.1 (OK)
sphinx >=0.6.6    :  1.8.3 (OK)
rope >=0.9.4      :  0.10.7 (OK)
jedi >=0.9.0      :  0.13.2 (OK)
psutil >=0.3      :  5.5.0 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
pandas >=0.13.1   :  0.24.0 (OK)
numpy >=1.7       :  1.15.4 (OK)
sympy >=0.7.3     :  None (NOK)
cython >=0.21     :  None (NOK)
qtconsole >=4.2.0 :  4.4.3 (OK)
IPython >=4.0     :  7.2.0 (OK)
matplotlib >=2.0.0:  3.0.2 (OK)
pylint >=0.25     :  2.2.2 (OK)
```

**Comment by PCManticore** [comment]
Thanks for the report @ccamara ! I updated the title to be more obvious that it's about a crash rather than something else.

**Comment by ccamara** [comment]
Great, I have just started with python and hence I am new to all of this. Thank you!

**Comment by PCManticore** [comment]
This should be fixed with the release of astroid 2.3.2. Can you give it a go and let me know if you still encounter this issue? Doing a `pip install pylint==2.4.3 astroid==2.4.2` should do the trick.
